test/shell/test_misc.sh

```diff
@@ -55,12 +55,21 @@ test_repl() {
 
 test_benchmark_jmh() {
   RES=$(bazel run -- test/jmh:test_benchmark -i1 -f1 -wi 1)
-  RESPONSE_CODE=$?
+  if [ $? -ne 0 ]; then
+    exit 1
+  fi
   if [[ $RES != *Result*Benchmark* ]]; then
     echo "Benchmark did not produce expected output:\n$RES"
     exit 1
   fi
-  exit $RESPONSE_CODE
+
+  set +e
+
+  bazel build test_expect_failure/jmh:jmh_reports_failure
```

Comment:
Can you extract it to a new method? The cost is small and the value is pinpointing what functionality regressed. Additionally the test name can say what’s going on since the error message right now is a bit vague. A separate test gives it context

Reply:
Extracted it to `test_benchmark_jmh_failure`.